java/src/main/java/ai/onnxruntime/OrtProviderOptions.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.io.IOException;
+
+/** An abstract base class for execution provider options classes. */
+// Note this lives in ai.onnxruntime to allow subclasses to access the OnnxRuntime.ortApiHandle
+// package private field.
+public abstract class OrtProviderOptions implements AutoCloseable {
+  static {
+    try {
+      OnnxRuntime.init();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load onnx-runtime library", e);
+    }
+  }
+
+  protected final long nativeHandle;
+
+  /**
+   * Constructs a OrtProviderOptions wrapped around a native pointer.
+   *
+   * @param nativeHandle The native pointer.
+   */
+  protected OrtProviderOptions(long nativeHandle) {
+    this.nativeHandle = nativeHandle;
+  }
+
+  /**
+   * Allow access to the api handle pointer for subclasses.
+   *
+   * @return The api handle.
+   */
+  protected static long getApiHandle() {
```

Comment:
>getApiHandle [](http://example.com/codeflow?start=24&length=12)

This I do not understand why a options class needs to have an accessor to something global, already exposed by OnnxRuntime.

Reply:
Because the provider implementations are in `ai.onnxruntime.providers` and `OnnxRuntime.ortApiHandle` has package level access in `ai.onnxruntime` so it's not visible. This allows it to be visible to subclasses of `OrtProviderOptions` no matter what package they are in.

I was trying to not add too many classes to the main package, but it's rather contorted here.